openshift/Dockerfile.requirements

```diff
@@ -0,0 +1,59 @@
+FROM registry.ci.openshift.org/ocp/4.17:base-rhel9
+
+RUN yum install -y libffi-devel python3-devel gcc python3-pip
+    
+COPY ./Pipfile* ./
+
+# The build dependencies are required by cachito. Following script
+# does exactly the same. More info at: https://github.com/containerbuildsystem/cachito/blob/master/docs/pip.md#build-dependencies
+RUN curl -LO https://raw.githubusercontent.com/containerbuildsystem/cachito/master/bin/pip_find_builddeps.py \
+    && chmod +x pip_find_builddeps.py
+
+# Create requirements.in file from the pipenv created using the
+# same Pipfile and Pipfile.lock used for upstream image. Then
+# use pip-compile to generate the requirements.txt file. Copy
+# setuptools into requirements-build.txt as pip-compile will
+# omit it from the requierements.txt file.
+RUN python3 -m pip install pipenv==2023.11.15 \
+    && python3 -m pip install pip-tools \
+    && pipenv install --deploy \
+    && pipenv run pip freeze --all  > ./requirements.in \
+    # Add wheel and setuptools to requirements-build.in file as these packages
+    # are part of the build dependencies of some packages in requirements.in file.
+    && grep "wheel==" ./requirements.in >> ./requirements-build.in || true \
+    && grep "setuptools==" ./requirements.in >> ./requirements-build.in || true \
+    && pip-compile --output-file=./requirements.txt ./requirements.in --strip-extras \
+    # NOTE (arkadeepsen): Comment out ansible-runner, ansible-runner-http and
+    # python-daemon as both ansible-runner and ansible-runner-http has
+    # python-daemon as dependency. pip_find_builddeps.py encounters an error
+    # when trying to get the build dependencies of python-daemon==3.0.1.
+    # TODO: Whenever a newer version of python-daemon is released check
+    # whether pip_find_builddeps.py is able to fetch its build dependencies
```

Comment:
The `pip_find_builddeps.py` script fails for `python-daemon==3.0.1`. I am not sure any future version of `python-daemon` will not have the same issues. Do you have any suggestions how this can be automated?

Also, the requirements files only needs to be regenerated when there's a change in the upstream Pipfile/Pipfile.lock files. Otherwise, we should be good with the python package dependencies.